src/components/molecules/PostFilter/index.tsx

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import { PageSearchParams } from "@/types/pageSearchParams";
+import { usePathname, useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+
+function PostFilter({ searchParams }: PageSearchParams) {
+  const pathname = usePathname();
+  const [searchParamsState, setSearchParamsState] = useState(new URLSearchParams(searchParams));
+  const [isAll, setIsAll] = useState(searchParamsState.get("all") !== "false");
+  const router = useRouter();
+
+  useEffect(() => {
+    setSearchParamsState(new URLSearchParams(searchParams));
+  }, [searchParams]);
+  useEffect(() => {
+    setIsAll(searchParamsState.get("all") !== "false");
+  }, [searchParamsState]);
```

Comment:
props로 받은 searchParams에 의해 변경되는 searchParamsState를 상태로 만드셨는데
Props가 변경되면 컴포넌트 내부 코드는 재실행되기 때문에 상태로 만드실 필요 없이 그냥 변수로 선언하셔도 됩니다!

```suggestion
  const searchParamsState = new URLSearchParams(searchParams);
  const isAll = searchParamsState.get("all") !== "false";
  const router = useRouter();
```

Reply:
리팩토링 완료했습니다.